OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -135,6 +136,7 @@ public ProductionQueue(ActorInitializer init, Actor playerActor, ProductionQueue
 
 		void INotifyCreated.Created(Actor self)
 		{
+			playerPower = playerActor.TraitOrDefault<PowerManager>();
```

Comment:
You should be able to use `self.Owner.PlayerActor` here and drop the `playerActor` variable.

Reply:
self can be a player actor if the queue is attached to a player, so using .Owner.PlayerActor will crash in this case.

Reply:
This can be solved as per https://github.com/OpenRA/OpenRA/blob/8a6091884127ea2c6706662332ee70097c81885e/OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs#L64-L77